### ISSUE TITLE
Unable to set heart-beat header using failoverUri string

### ISSUE THREAD

**Issue by choudhary001** [gdaws/node-stomp#53]
Hi,

I am trying to set heart-beat header for node stomp client using failover uri string but it is not picking up correct header value.

I am using below url: (case#1)
```javascript
var connectHeaderString = "connect[connectHeaders][client-id]="+ clientId + "&connect[connectHeaders][heart-beat]=44444,10000";
var failoverUri = "failover:(stomp://argon45.nyc:61613,stomp://argon45.nyc:61613)?"
                  + connectHeaderString;
```
In this case the heartbeat value isn't considered a string.

It doesn't work if I try this also (case#2)
```
var connectHeaderString = "connect[connectHeaders][client-id]="+ clientId + "&connect[connectHeaders][heart-beat]='44444,10000'";
```
In this case the regex doesn't match the heart beat header as it header value contains the extra quotes:
`heart-beat is string type - "'10000,10000'"`

I checked by adding log statements around heart beat code in Client.js:
```javascript
var heartbeat = this.getHeartbeat();
  console.log("heart-beat before setting" + JSON.stringify(heartbeat));

  //console.log('hearders: \n' + JSON.stringify(headers));
  if (typeof headers['heart-beat'] === "string") {
    console.log('heart-beat is string type - ' + JSON.stringify(headers['heart-beat']));
    var match = headers['heart-beat'].match(/^(\d+) *, *(\d+)$/);
    if (match) {
      console.log('heart-beat matches the regex')
      heartbeat = [parseInt(match[1], 10), parseInt(match[1], 10)];
      this.setHeartbeat(heartbeat);
      console.log("heart-beat after setting" + JSON.stringify(heartbeat));
    }
  }

  headers['heart-beat'] = heartbeat[0] + "," + heartbeat[1];
  console.log('heart-beat set to - ' + JSON.stringify(headers['heart-beat']));
```

The logs i get on console are below (for case#1):
```
Connecting to argon45.nyc:61613
heart-beat before setting[0,0]
heart-beat set to - "0,0"
Connected to argon45.nyc:61613, with options: {"host":"argon45.nyc","port":61613,"connectHeaders":{"client-id":"stompit-test-consumer","heart-beat":44444},"ssl":false}
```

Logs for case#2:
```
Connecting to argon45.nyc:61613
heart-beat before setting[0,0]
heart-beat is string type - "'44444,10000'"
heart-beat set to - "0,0"
Connected to argon45.nyc:61613, with options: {"host":"argon45.nyc","port":61613,"connectHeaders":{"client-id":"stompit-test-consumer","heart-beat":"'44444,10000'"},"ssl":false}
```

**Comment by gdaws** [comment]
I suspect you're using an old version of Stompit. Upgrading to version 0.23 should hopefully resolve your issue.

When calling parseFailoverUri from the node REPL I see that the query string is correctly parsed:

```
> parseFailoverUri(uri).options.connect
{ connectHeaders: { 'client-id': 'stompit-test-consumer', 'heart-beat': '44444,10000' } }
```

Prior to version 0.23 a query string parameter value of "44444,10000" is parsed into a number value of 44444.

**Comment by choudhary001** [comment]
Hi,

Upgraded to v0.23 but it also has an issue as it is passing first value in heart-beat parameter as both client and server value.
In my case, I only want the server to send heart-beats so i specify 0,10000 but because of heart-beat code in Client.js it is passing 0,0. See [https://github.com/gdaws/node-stomp/blob/master/lib/Client.js#L129](https://github.com/gdaws/node-stomp/blob/master/lib/Client.js#L129) 

As per STOMP v1.1 spec - if both client and server values are specified, max value out these two should be used - [link](http://stomp.github.io/stomp-specification-1.1.html#Heart-beating)

**Client side:**
Connecting to argon45.nyc:61613
heartbeat before setting - [0,0]
matched string - ["50,100","50","100"]
matched heartbeat pattern - [50,50]
heartbeat after setting - "50,50"
Connected to argon45.nyc:61613, with options: {"host":"argon45.nyc","port":61613,"connectHeaders":{"client-id":"stompit-test-consumer-gold04.nyc.deshaw.com","heart-beat":"50,100"},"ssl":false}


**Server:**
2016-10-26 03:47:38,112 [MQ NIO Worker 2] TRACE StompIO                        - Received: 
CONNECT
accept-version:1.0,1.1,1.2
heart-beat:50,50
host:argon45.nyc
client-id:stompit-test-consumer

^@
2016-10-26 03:47:38,113 [MQ NIO Worker 2] DEBUG StompInactivityMonitor         - Stomp Inactivity Monitor read check interval: 50ms, write check interval: 50ms
2016-10-26 03:47:38,114 [MQ NIO Worker 2] DEBUG ProtocolConverter              - Stomp Connect heartbeat conf RW[50,50]
2016-10-26 03:47:38,121 [MQ NIO Worker 2] TRACE StompIO                        - Sending: 
CONNECTED
server:ActiveMQ/5.11.1
heart-beat:50,50
session:stompit-test-consumer
version:1.2

It sends keep alive every 50ms 
2016-10-26 03:47:38,216 [yMonitor Worker] TRACE StompIO                        - Sending: 
KEEPALIVE

^@
2016-10-26 03:47:38,266 [yMonitor Worker] TRACE StompIO                        - Sending: 
KEEPALIVE

^@
2016-10-26 03:47:38,316 [yMonitor Worker] TRACE StompIO                        - Sending: 
KEEPALIVE

**Comment by gdaws** [comment]
This test 37b8b0c2a0c887b942653eaa832c5c4d15e3b205 confirms a heart-beat cx,cx bug and the fix is at 8b01bd57f15d6ced12cf18210621952897a4294d. 

A new version of stompit v0.24.0 has been published to the npm registry.

Thanks for reporting the issue.
